### PR TITLE
Create doc on how to setup an application on S3

### PR DESCRIPTION
**Pull request by gideonthomas**

### CODE REVIEW

File: docs/aws/deploy-s3.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+4. __Setup Hosting__: Next you need to setup hosting for the application. Here's how you do it:
+  1. Select the "Properties" tab and expand the "Static Website Hosting" section.
+  2. Select "Enable website hosting" and type in your application entry-point (Index Document) and the default error page you would like to show (set it to the same value of the Index Document if you don't have a custom error page).
+  3. Click "Save". You should now see an "Endpoint" url that you can access your bucket's contents at. However, this url won't be accessible until you setup permissions for the bucket as described in the next step.
+5. __Permissions__: Adequate permissions need to be set on the bucket to enable access to it. The steps for this are:
+  1. Expand the "Permissions" section in the "Properties" tab.
+  2. Select "Add Bucket Policy" and click on the "AWS Policy Generator" link in the bottom-left corner. This will open a new window/tab with the Policy Generator page.
+  3. In the Policy Generator page:
+    1. Select the "Policy Type" as "S3 Bucket Policy". 
+    2. Set the "Principal" to "*"(without quotes) to allow access to all the content in the bucket.
```
Comment (cadecairos) [inline docs/aws/deploy-s3.md]: The principal actually refers to the entity making the request for content. Using "*" here means the policy applies to everyone.

File: docs/aws/deploy-s3.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  3. Click "Save". You should now see an "Endpoint" url that you can access your bucket's contents at. However, this url won't be accessible until you setup permissions for the bucket as described in the next step.
+5. __Permissions__: Adequate permissions need to be set on the bucket to enable access to it. The steps for this are:
+  1. Expand the "Permissions" section in the "Properties" tab.
+  2. Select "Add Bucket Policy" and click on the "AWS Policy Generator" link in the bottom-left corner. This will open a new window/tab with the Policy Generator page.
+  3. In the Policy Generator page:
+    1. Select the "Policy Type" as "S3 Bucket Policy". 
+    2. Set the "Principal" to "*"(without quotes) to allow access to all the content in the bucket. 
+    3. For the "Action", you will need to at least select the "GetObject" permission. If you want to allow uploads to the S3 bucket from anywhere outside AWS, you should also enable the "PutObject" permission.
```
Comment (cadecairos) [inline docs/aws/deploy-s3.md]: noooooo never allow putObject with "*" as the principal, since this lets anybody upload files. 

For uploading outside of AWS you'd create a new IAM user with a policy that allows access to the S3 bucket with the putObject action.

File: docs/aws/deploy-s3.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    4. For the "ARN" set it to your bucket ARN which will look something like `arn:aws:s3:::<bucket_name>`
```
Comment (cadecairos) [inline docs/aws/deploy-s3.md]: The arn needs a wildcard: `arn:aws:s3:::<bucket_name>/*`

File: docs/aws/deploy-s3.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    5. Click on "Add Statement" and then "Generate Policy". A JSON document is created for you. Copy the text from there.
+  4. Paste the JSON permission object into the text area back in the "Add Bucket Policy" dialog in the previous page and click "Save".
+  5. Verify that you can now access the bucket's content by using the url seen in the "Static Website Hosting" section.
+6. Your S3 bucket should now be setup correctly.
+
+### Setting up a CloudFront distribution
+Once you have your S3 bucket set up, you need to add a CloudFront CDN to serve the bucket's content effectively. The steps for doing so are as follows:
+
+1. __Role__: Before proceeding, make sure you switch to the appropriate role so that the role used for CloudFront is the same as the role in which the SSL certificate you would like to use exists.
```
Comment (cadecairos) [inline docs/aws/deploy-s3.md]: lets try to reword this to use the word role less.

"Before proceeding, make sure you switch to the appropriate role so that the CloudFront distribution you create is in the same account where the SSL certificate you'd like to use is stored"

Or something like that

**Comment by cadecairos** [comment]
Just a few things to fix! Awesome docs Gideon!

File: docs/aws/deploy-s3.md (hunk 1 above)
Comment (gideonthomas) [inline docs/aws/deploy-s3.md]: so should I add "for everyone" at the end?

File: docs/aws/deploy-s3.md (hunk 2 above)
Comment (gideonthomas) [inline docs/aws/deploy-s3.md]: I'll add a note

File: docs/aws/deploy-s3.md (hunk 3 above)
Comment (gideonthomas) [inline docs/aws/deploy-s3.md]: Are you sure? I don't see it in the config

Comment (cadecairos) [inline docs/aws/deploy-s3.md]: You want the policy to apply to objects in the bucket, not the bucket itself

**Comment by cadecairos** [comment]
@gideonthomas did you fix this up yet?

**Comment by gideonthomas** [comment]
ah damn, didn't commit my changes. Doing now.

**Comment by gideonthomas** [comment]
@cadecairos how about now?

**Comment by cadecairos** [comment]
Looks good, merging
